src/mslice/plotting/plot_window/slice_plot.py

```diff
@@ -331,40 +332,43 @@ def update_canvas(self, cbar_range, cbar_log, x_range, y_range, title):
         self._canvas.draw()
 
     def _run_temp_dependent(self, slice_plotter_method, previous):
-        temp_value_raw = None
-        temp_value = None
         try:
             slice_plotter_method(self.ws_name)
-        except ValueError:  # sample temperature not yet set
-            try:
-                temp_value_raw, field = self.ask_sample_temperature_field(str(self.ws_name))
-            except RuntimeError:  # if cancel is clicked, go back to previous selection
-                self.set_intensity(previous)
-                return False
-            if field:
-                self._slice_plotter_presenter.add_sample_temperature_field(temp_value_raw)
-                temp_value = self._slice_plotter_presenter.update_sample_temperature_from_field(self.ws_name)
-            else:
-                temp_value = get_sample_temperature_from_string(temp_value_raw)
-                if temp_value is not None:
-                    try:
-                        temp_value = float(temp_value)
-                    except ValueError:
-                        temp_value = None
-            if temp_value is None or temp_value < 0:
-                self.plot_window.display_error("Invalid value entered for sample temperature. Enter a value in Kelvin \
-                                                or a sample log field.")
-                self.set_intensity(previous)
+        except ValueError:  # sample temperature not yet set, get it and reattempt method
+            if self._set_sample_temperature(previous):
+                slice_plotter_method(self.ws_name)
```

Comment:
I think it is possible for `previous` to be None, and so there could still be an unhandled exception on line 339?

Reply:
This cannot be none, its actually the `QAction` related to the previously selected intensity type in the drop down menu. As a menu item is always selected, there is always a previous.